rswag-specs/lib/rswag/specs/swagger_formatter.rb

```diff
@@ -124,19 +124,44 @@ def upgrade_response_produces!(swagger_doc, metadata)
         # Accept header
         mime_list = Array(metadata[:operation][:produces] || swagger_doc[:produces])
         target_node = metadata[:response]
-        upgrade_content!(mime_list, target_node)
+        target_node.merge!(content: {})
+        upgrade_schema!(mime_list, target_node)
+        upgrade_examples!(mime_list, target_node)
         metadata[:response].delete(:schema)
+        metadata[:response].delete(:example)
+        metadata[:response].delete(:examples)
       end
 
-      def upgrade_content!(mime_list, target_node)
-        target_node.merge!(content: {})
+      def upgrade_schema!(mime_list, target_node)
         schema = target_node[:schema]
         return if mime_list.empty? || schema.nil?
 
+        set_mime_list_contents!(target_node, mime_list, :schema, schema)
+      end
+
+      def upgrade_examples!(mime_list, target_node)
```

Comment:
@pedrocarmona I pulled your code and wired it to my own project - as it is, schemas generated from your fork of rswag with examples added in this way will fail openapi_cli validation, because you're dumping the contents of the old-style [metadata][examples] into [metadata][content][mime_type][examples] without handling the contents.

If the rswag-consuming application implements the suggested code for automatic response examples (which is:
```
after do |example|
  example.metadata[:response][:examples] = { 'application/json' => JSON.parse(response.body, symbolize_names: true) }
end
```
per https://github.com/rswag/rswag#enable-auto-generation-examples-from-responses)

then their schema has [metadata][examples][mime-type]. Your code then dumps this into [metadata][content][mimetype][examples][mime-type][actual example].

You end up with a result that looks like:
```
responses:
        '200':
          description: returns kubeconfig data
          content:
            application/json:
              examples:
                application/json:
                  some_actual_response_content: "asdasdasda"
```

which, on running the openapi validator (I'm running it through docker as per https://openapi-generator.tech/docs/installation/#docker with `docker run --rm -v [APP_ROOT]/local openapitools/openapi-generator-cli:v4.2.3 validate --input-spec /local/[MY_SCHEMA_DIR]` if you want 100% reproducibility)

you get reported schema errors like:
```
Validating spec (/local/openapi/v1/kubemgr.yaml)
Errors:
        -attribute paths.'/spaces'(post).responses.200.content.examples.application/json.[CONTENTS] is unexpected
```

Fixing this is going to be tricky, and I'm not sure what the right path forward is. Setting the mime list contents needs to be less naive about the contents if it's important to keep 100% backwards compatibility with the way the README says to add examples.

Reply:
A solution that I implemented in my own fork (based off your fork now):
changing swagger_formatter.rb method `upgrade_examples!` to:

```
def upgrade_examples!(mime_list, target_node)
        example = target_node[:example]
        examples = target_node[:examples]

        if example
          set_mime_list_contents!(target_node, mime_list, :example, example)
        elsif examples
          #check if examples contains mime_types from swagger2.0 response example style
          non_mime_type_examples = {}
          examples.each do |key, value|
            if mime_list.include? key
              set_mime_type_content!(target_node, key, :examples, value)
            else
              non_mime_type_examples[key] = value
            end
          end
          set_mime_list_contents!(target_node, mime_list, :examples, non_mime_type_examples)
        end
      end
```

I don't know if this solution will work universally, but it should:
* Allow the consumer to use the current README suggested method of adding autogenerated response examples.
* Override those autogenerated examples with integration-spec-level-defined custom example responses
* Do so in a way that is compliant with the openapi-generator-cli validator.

Would it be valuable for me to open a PR to your fork with this, or...open a PR here, or something else?

Reply:
yes, I think I will need to add a bit more tests on top of your PR.

Reply:
Opened a PR on your fork instead of here? I figured that would be faster + more helpful, if you'd rather I open it here above yours instead lmk

Reply:
merged